src/DictVectors/initiators.jl

```diff
@@ -0,0 +1,219 @@
+    threshold::V
+end
+
+function value(i::CoherentInitiator, v::InitiatorValue)
+    if !iszero(v.initiator) || abs(v.unsafe) > i.threshold
+        return v.initiator + v.safe + v.unsafe
+    else
+        return v.initiator + v.safe
+    end
+end
+
+"""
+    InitiatorDVec
+
+Functionally identical to [`DVec`](@ref), but contains [`InitiatorValue`]s
+"""
+struct InitiatorDVec{
+    K,V,D<:Dict{K,InitiatorValue{V}},S<:StochasticStyle{V},I<:AbstractInitiator
+} <: AbstractDVec{K,V}
+    storage::D
+    style::S
+    initiator::I
+end
+
+###
+### Constructors
+###
+function InitiatorDVec(
+    dict::Dict{K,V}; style=default_style(V), initiator=Initiator(one(V)), capacity=0
+) where {K,V}
+    T = eltype(style)
+    storage = Dict{K,InitiatorValue{T}}()
+    sizehint!(storage, capacity)
+    for (k, v) in pairs(dict)
+        storage[k] = InitiatorValue{T}(safe=v)
+    end
+    return InitiatorDVec(storage, style, initiator)
+end
+function InitiatorDVec(args...; kwargs...)
+    storage = Dict(args...)
+    return InitiatorDVec(storage; kwargs...)
+end
+function InitiatorDVec(
+    dv::AbstractDVec{K,V}; style=StochasticStyle(dv), initiator=Initiator(one(V)), capacity=0
+) where {K,V}
+    storage = Dict{K,InitiatorValue{V}}()
+    sizehint!(storage, max(capacity, length(dv)))
+    for (k, v) in pairs(dv)
+        storage[k] = InitiatorValue{V}(safe=v)
+    end
+    return InitiatorDVec(storage, style, initiator)
+end
+function InitiatorDVec{K,V}(; kwargs...) where {K,V}
+    return InitiatorDVec(Dict{K,V}(); kwargs...)
+end
+
+function Base.empty(dvec::InitiatorDVec)
+    return InitiatorDVec(empty(dvec.storage), dvec.style, dvec.initiator)
+end
+function Base.empty(dvec::InitiatorDVec, args...)
+    return InitiatorDVec(empty(dvec.storage, args...); initiator=dvec.initiator)
+end
+Base.similar(dvec::InitiatorDVec, args...; kwargs...) = empty(dvec, args...; kwargs...)
+
+###
+### Show
+###
+function Base.summary(io::IO, dvec::InitiatorDVec{K,V,S}) where {K,V,S}
+    len = length(dvec)
+    print(io, "InitiatorDVec{$K,$V,$S} with $len entries")
+end
+
+###
+### Interface
+###
+StochasticStyle(dv::InitiatorDVec) = dv.style
+storage(dv::InitiatorDVec) = dv.storage
+
+function Base.getindex(dvec::InitiatorDVec{<:Any,V}, add) where {V}
+    return value(dvec.initiator, get(dvec.storage, add, zero(InitiatorValue{V})))
+end
+function Base.setindex!(dvec::InitiatorDVec{<:Any,V}, v, k) where {V}
+    if iszero(v)
+        delete!(dvec.storage, k)
+    else
+        dvec.storage[k] = InitiatorValue{V}(safe=v)
+    end
+    return v
+end
+
+function Base.pairs(dvec::InitiatorDVec)
+    return Iterators.map(pairs(dvec.storage)) do ((k, v))
+        k => value(dvec.initiator, v)
+    end
+end
+function Base.values(dvec::InitiatorDVec)
+    return Iterators.map(values(dvec.storage)) do v
+        value(dvec.initiator, v)
+    end
+end
+function Base.get(dvec::InitiatorDVec, args...)
+    return(value(dvec.initiator, get(dvec.storage, args...)))
```

Comment:
Is this the same as
```suggestion
    return value(dvec.initiator, get(dvec.storage, args...))
```
?

Reply:
Yes. Don't know how that got there.